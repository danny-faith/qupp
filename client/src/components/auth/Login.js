import React, { Component } from 'react'
import { Row, Col, Button, Input } from 'react-materialize';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import classnames from 'classnames';
import { loginUser } from '../../actions/authActions';

class Login extends Component {
  state = {
    email: '',
    password: '',
    errors: {}
  }
  onChange = e => {
    this.setState({[e.target.name]: e.target.value});
  }
  onSubmit = e => {
    e.preventDefault();

    const userData = {
      email: this.state.email,
      password: this.state.password
    }
    this.props.loginUser(userData);
	}
	componentDidMount = () => {
		if (this.props.auth.isAuthenticated) {
				this.props.history.push('/dashboard');
		}
	}
  componentWillReceiveProps = (nextProps) => {
    if (nextProps.auth.isAuthenticated) {
      this.props.history.push('/dashboard');
    }
		if (nextProps.errors) {
			this.setState({errors: nextProps.errors});
		}
	}
  render() {
		const { errors } = this.state;

    return (
      <Row>
				<Col s={6} className="offset-s3">
					<h1 className="center">Login</h1>
					<form noValidate onSubmit={this.onSubmit}>
            <Row>
							<Col s={12}>
								<Input
									id={"email"}
									className={classnames({
										'invalid': errors.email
									})} 
									type="text"
									name="email"
									placeholder="Email"
									s={12}
									label="Email"
									onChange={this.onChange}
									value={this.state.email}
									/>
									{errors.email && (<p className="red-text col no-margin">{errors.email}</p>)}
							</Col>
						</Row>
            <Row>
							<Col s={12}>
								<Input
									id={"password"}
									className={classnames({
										'invalid': errors.password
									})} 
									type="password"
									name="password"
									placeholder="Password"
									s={12}
									label="Password"
									onChange={this.onChange}
                  value={this.state.password}
									/>
									{errors.password && (<p className="red-text col no-margin">{errors.password}</p>)}
							</Col>
						</Row>
            <Row>
							<Col s={12}>
								<Button className="btn-small right" waves="light">Login</Button>
							</Col>
						</Row>
          </form>
         </Col>
      </Row>
    )
  }
}

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
  auth: state.auth,
  errors: state.errors
});

export default connect(mapStateToProps, { loginUser })(Login);
