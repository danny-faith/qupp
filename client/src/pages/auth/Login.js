import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Row, Col, Button } from 'react-materialize';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import TextFieldGroup from '../../components/common/TextFieldGroup';
import { loginUser } from '../../actions/authActions';

class Login extends Component {
	state = {
		usernameOrEmail: '',
		password: '',
		errors: {}
	}
	onChange = e => {
		const errors = this.state.errors;
		errors[e.target.name] = '';
		this.setState({
			[e.target.name]: e.target.value,
			errors
		});
	}
  	onSubmit = e => {
    	e.preventDefault();

		const userData = {
			usernameOrEmail: this.state.usernameOrEmail,
			password: this.state.password
		}
    	this.props.loginUser(userData);
	}
	componentDidMount = () => {
		if (this.props.auth.isAuthenticated) {
			this.props.history.push('/dashboard');
		}
	}
  	componentWillReceiveProps = (nextProps) => {
		if (nextProps.auth.isAuthenticated) {
			this.props.history.push('/dashboard');
		}
		if (nextProps.errors) {
			this.setState({errors: nextProps.errors});
		}
	}
  	render() {
		const { errors } = this.state;

		return (
			<Row>
				<Col s={12} m={6} className="offset-m3">
					<h2 className="center">Login</h2>
					<form noValidate onSubmit={this.onSubmit}>
						<Row>
							{/* <TextFieldGroup
								placeholder="Username or email"
								name="usernameOrEmail"
								type="text"
								value={this.state.usernameOrEmail}
								onChange={this.onChange}
								error={errors.usernameOrEmail}
							/>
							<TextFieldGroup
								placeholder="Password"
								name="password"
								type="password"
								value={this.state.password}
								onChange={this.onChange}
								error={errors.password}
							/> */}
							<Col s={12}>
								<Link className='col s12' to="/forgotten-password">Forgotten password?</Link>
								<Button className="btn-small right" waves="light">Login</Button>
							</Col>
						</Row>
					</form>
				</Col>
			</Row>
		)
  	}
}

Login.propTypes = {
	loginUser: PropTypes.func.isRequired,
	auth: PropTypes.object.isRequired,
	errors: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
	auth: state.auth,
	errors: state.errors
});

export default connect(mapStateToProps, { loginUser })(Login);
