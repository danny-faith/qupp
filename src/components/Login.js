import React, { Component } from 'react';
import axios from 'axios';
import { Row, Col, Button, Input } from 'react-materialize';

class Login extends Component {
    usernameInput = React.createRef();
    passwordInput = React.createRef();
    handleFormSubmit = event => {
        event.preventDefault();
        const username = this.usernameInput.current.state.value;
        const password = this.passwordInput.current.state.value;

        console.log('username: ', username);
        console.log('password: ', password);
        const params = {
            username: username,
            password: password
        }
        axios({
            method: 'post',
            url: 'http://localhost:3333/login',
            headers: {
                "Content-Type": "application/x-www-form-urlencoded"
            },
            params: params
        })
        .then(res => {
          console.log(res);
          // SHOULD BE ERROR CATCHING IN HERE!!!
        })
        .catch(function (error) {
            console.log('error: ', error);
            console.log('error.config: ', error.config);
        });
    }
    render() {
        return (
            <Row>
                <form onSubmit={this.handleFormSubmit}>
                    <Col s={12}>
                        <Input id={"userName"} defaultValue='daniel8' ref={this.usernameInput} placeholder="is this even used" s={12} label="Username" />
                    </Col>
                    <Col s={12}>
                        <Input id={"password"} defaultValue='password' ref={this.passwordInput} placeholder="is this even used" s={12} label="Password" />
                    </Col>
                    <Col s={12}>
                        <Button waves="light" className="pink lighten-2">Login</Button>
                    </Col>
                </form>
            </Row>
        )
    }
}

export default Login;
