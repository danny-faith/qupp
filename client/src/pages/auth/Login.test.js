import React from 'react';
import { shallow, render, mount } from 'enzyme';
import { Login } from './Login';

describe('Login component', () => {
    let wrapper;
    const mockLoginUserfn = jest.fn();
    const auth = {
        isAuthenticated: true,
    }
    beforeEach(() => {
        wrapper = shallow(
        <Login
            loginUser={mockLoginUserfn} 
            auth={auth}
            errors={{}}
            history={[]}
        />)
    });

    
    it('should shallow render correctly', () => {
        const auth = {
            isAuthenticated: false,
        }
        const loginUser = jest.fn();
        const component = shallow(
            <Login
                auth={auth}
                loginUser={loginUser}
                errors={{}}
                history={[]}
            />
          );
          expect(component).toMatchSnapshot();
    });
    
    describe('When the form is submitted', () => {
        it('should call the mock login function', () => {
        // Ok to use `form` as a selector as there is only one form within the component
         wrapper.find('form').simulate(
            'submit', 
            {preventDefault() {}}
        );
        expect(mockLoginUserfn.mock.calls.length).toBe(1)
        })
    });

    it('should be called with the email and password in the state as arguments', () => {
        // fill in email field with blah@gmail.com     
        wrapper.find('[name="usernameOrEmail"]').simulate(
          'change', 
          {target: 
            {name: 'usernameOrEmail', value: 'blah@gmail.com'}
          }
        )
        // fill in password field with cats  
        wrapper.find('[name="password"]').simulate(
          'change', 
          {target: 
             {name: 'password', value: 'cats'}
          }
        )
        // simulate form submission   
        // Ok to use `form` as a selector as there is only one form within the component
        wrapper.find('form').simulate(
          'submit', 
          {preventDefault() {}}
        )
        // test to see arguments used after its been submitted 
        expect(mockLoginUserfn.mock.calls[1][0]).toEqual(
          {usernameOrEmail: 'blah@gmail.com', password: 'cats'}
        )
    });
});