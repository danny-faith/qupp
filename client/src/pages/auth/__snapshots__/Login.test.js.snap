// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should shallow render correctly 1`] = `
<Row>
  <Col
    className="offset-m3"
    m={6}
    s={12}
  >
    <h2
      className="center"
    >
      Login
    </h2>
    <form
      noValidate={true}
      onSubmit={[Function]}
    >
      <Row>
        <TextFieldGroup
          label="Username or email"
          name="usernameOrEmail"
          onChange={[Function]}
          type="text"
          value=""
        />
        <TextFieldGroup
          label="Password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <Col
          s={12}
        >
          <Link
            className="col s12"
            replace={false}
            to="/forgotten-password"
          >
            Forgotten password?
          </Link>
          <Button
            className="btn-small right"
            node="button"
            waves="light"
          >
            Login
          </Button>
        </Col>
      </Row>
    </form>
  </Col>
</Row>
`;
